src/protocol/mod.rs

```diff
@@ -105,16 +105,8 @@ impl<T> Transport for TarpcTransport<T>
     }
 
     fn write(&mut self, req: Frame<Packet, io::Error>) -> io::Result<Option<()>> {
-        match req {
-            Frame::Message(msg) => {
-                self.outbound.push_back(msg);
-                self.flush()
-            }
-            Frame::MessageWithBody(..) => unreachable!(),
-            Frame::Body(_) => unreachable!(),
-            Frame::Error(_) => unreachable!(),
-            Frame::Done => unreachable!(),
-        }
+        self.outbound.push_back(req.unwrap_msg());
```

Comment:
`unwrap_msg` is a helper fn that does essentially the same thing as what the match was doing.